carbonj.service/src/main/java/com/demandware/carbonj/service/engine/GraphiteMetricSearchServlet.java

```diff
@@ -120,6 +129,35 @@ protected void doGet( HttpServletRequest req, HttpServletResponse res )
             res.getWriter().write( gson.toJson( metrics ) );
             res.getWriter().close();
         }
+        else if ( msgpack )
+        {
+            ObjectMapper objectMapper = new ObjectMapper( new MessagePackFactory() );
+
+            List<MsgPackMetric> msgPackMetrics = new ArrayList<MsgPackMetric>();
+
+            for ( Metric metric : metrics )
+            {
+                msgPackMetrics.add( new MsgPackMetric( metric ) );
+            }
+
+            OutputStream output = res.getOutputStream();
+            try
+            {
+                // Serialize the metrics
+                byte[] serialized = objectMapper.writeValueAsBytes( msgPackMetrics );
+                res.setContentLength( serialized.length );
+                output.write( serialized );
+
+            }
+            catch ( IOException e )
+            {
+                LOG.error( "carbonapi request: error writing msgpack response", e.getMessage() );
+            }
+            finally
+            {
+                output.close();
```

Comment:
You could get rid of the finally block by embedding output into try-with-resources statement.